libs/math/include/math/tensor.hpp

```diff
@@ -2585,10 +2588,13 @@ template <typename T, typename C>
 typename Tensor<T, C>::SliceIteratorType Tensor<T, C>::TensorSlice::begin()
 {
   auto ret = SliceIteratorType(this->tensor_, this->range_);
-  if (this->axis_ != 0)
+
+  // If there is only one axis and is 0, it's already at front
+  if (this->axis_.size() != 1 || this->axis_.at(0) != 0)
```

Comment:
please put extra parenthesis around the second condition (otherwise it can be slightly unclear whether the != should be executed before or after the ||